src/routes/manage_job/get.rs

```diff
@@ -0,0 +1,180 @@
+use crate::errors::ReacherError;
+use crate::routes::check_email::header::check_header;
+use sqlx::{Pool, Postgres};
+use warp::Filter;
+
+use serde::Serialize;
+
+use check_if_email_exists::CheckEmailOutput;
+use sqlx::types::chrono::{DateTime, Utc};
+use sqlx::types::Json;
+
+#[derive(sqlx::Type, Debug, Serialize, PartialEq, Eq)]
+#[sqlx(type_name = "valid_status", rename_all = "lowercase")]
+pub enum ValidStatus {
+	Running,
+	Completed,
+	Stopped,
+}
+
+/// Job record stores the information about a submitted job
+///
+/// `job_status` field is an update on read field. It's
+/// status will be derived from counting number of
+/// completed email verification tasks. It will be updated
+/// with the most recent status of the job.
+#[derive(sqlx::FromRow, Debug, Serialize)]
+pub struct JobRecord {
+	id: i32,
+	created_at: DateTime<Utc>,
+	total_records: i32,
+	job_status: ValidStatus,
+}
+
+/// Email record stores the result of a completed email verification task
+///
+/// It related to it's parent job through job_id
+/// It stores the result of a verification as jsonb field
+/// serialized from `CheckEmailOutput`
+#[derive(sqlx::FromRow, Debug)]
+pub struct EmailRecord {
+	job_id: i32,
+	email_id: String,
+	result: Json<CheckEmailOutput>,
+}
+
+/// Summary of a bulk verification job status
+#[derive(Debug, Serialize)]
+pub struct JobStatusSummaryResponseBody {
+	total_safe: i32,
+	total_risky: i32,
+	total_invalid: i32,
+	total_unknown: i32,
+}
+
+/// Complete information about a bulk verification job
+#[derive(Debug, Serialize)]
+pub struct JobStatusResponseBody {
+	job_id: i32,
+	created_at: DateTime<Utc>,
+	total_records: i32,
+	total_processed: i32,
+	summary: JobStatusSummaryResponseBody,
+	job_status: ValidStatus,
+}
+
+async fn job_status(
+	job_id: i32,
+	conn_pool: Pool<Postgres>,
+) -> Result<impl warp::Reply, warp::Rejection> {
+	let mut job_rec = sqlx::query_as!(
+		JobRecord,
+		r#"
+		SELECT id, created_at, total_records, job_status as "job_status: _" FROM blk_vrfy_job
+		WHERE id = $1
+		LIMIT 1
+		"#,
+		job_id
+	)
+	.fetch_one(&conn_pool)
+	.await
+	.map_err(|e| {
+		log::error!(
+			target:"reacher/v0/bulk/",
+			"Failed to get job record for [job_id={}] with [error={}]",
+			job_id,
+			e
+		);
+		ReacherError::from(e)
+	})?;
+
+	let agg_info = sqlx::query!(
+		r#"
+		SELECT
+			COUNT(*) as total_processed,
+			COUNT(CASE WHEN result ->> 'is_reachable' LIKE 'safe' THEN 1 END) as safe_count,
```

Comment:
According to some blog posts it's pretty performant - https://medium.com/geekculture/postgres-jsonb-usage-and-performance-analysis-cdbd1242a018.

We can even define indexes on subfields in the JSONB data. Since you're going for a lean db design, we can go with this and later experiment with separate columns and it's impact on performance.

Reply:
Sounds good!